src/EditorFeatures/Core/Implementation/CommentSelection/CommentUncommentSelectionCommandHandler.cs

```diff
@@ -171,51 +194,84 @@ internal void CollectEdits(ICommentUncommentService service, NormalizedSnapshotS
         /// <summary>
         /// Add the necessary edits to comment out a single span.
         /// </summary>
-        private void CommentSpan(ICommentUncommentService service, SnapshotSpan span, List<TextChange> textChanges, List<ITrackingSpan> trackingSpans)
+        private void CommentSpan(ICommentSelectionService service, SnapshotSpan span, List<TextChange> textChanges, List<ITrackingSpan> trackingSpans)
         {
-            var firstAndLastLine = DetermineFirstAndLastLine(span);
+            var (firstLine, lastLine) = DetermineFirstAndLastLine(span);
 
-            if (span.IsEmpty && firstAndLastLine.Item1.IsEmptyOrWhitespace())
+            if (span.IsEmpty && firstLine.IsEmptyOrWhitespace())
             {
+                // No selection, and on an empty line, don't do anything.
                 return;
             }
 
             if (!span.IsEmpty && string.IsNullOrWhiteSpace(span.GetText()))
             {
+                // Just whitespace selected, don't do anything.
                 return;
             }
 
-            if (span.IsEmpty || string.IsNullOrWhiteSpace(span.GetText()))
+            // Get the information from the language as to how they'd like to comment this region.
```

Comment:
this logic has been rewritten/cleaned up a bit.  So it should be reviewed somewhat closely.